File: habit-tracker/src/main/kotlin/com/habit/habit_tracker/dto/task/response/TaskResponse.kt
package com.habit.habit_tracker.dto.task.response

import com.fasterxml.jackson.annotation.JsonInclude
import com.habit.habit_tracker.enums.TaskDifficulty
import com.habit.habit_tracker.enums.TaskStatus
import java.time.LocalDateTime

@JsonInclude(JsonInclude.Include.NON_NULL)
data class TaskResponse(
    val id: Long,
    val userId: Long,
    val name: String,
    val description: String?,
    val status: TaskStatus,
    val difficulty: TaskDifficulty,
    val dueDate: LocalDateTime,
    val createdAt: LocalDateTime
)

File: client/src/types/dto/task/response/TaskResponse.ts
// src/types/task/TaskResponse.ts
import { TaskDifficulty } from "../../../enums/TaskDifficulty";
import { TaskStatus } from "../../../enums/TaskStatus";

export interface TaskResponse {
  id: number;
  userId: number;
  name: string;
  description?: string | null;
  status: TaskStatus;
  difficulty: TaskDifficulty;
  dueDate: string;
  createdAt: string;
}
